### PR TITLE
add a new platform i.MXRT1050-EVKB(Cortex-M7, based on ARMv7E-M)

### PR DESCRIPTION
**Pull request by Ellab-Yoon**
- ARMv7-M Architecture
- i.MXRT1050 Chip
- NXP MIMXRT1050-EVK Board Configuration
- TZSH Shell and Libraries

- qemu/tc_64k 
- qemu/build_test 
- artik053/grpc 
- artik053/minimal 
- artik053/kernel_sample 
- artik053/iotjs 
- artik053/tc 
- artik053

/st_things

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [dc0fdce7e3c46a28c6e5f2f7e321d3d11dcf2fb3] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [dc0fdce7e3c46a28c6e5f2f7e321d3d11dcf2fb3] - Code Rule Check Result:
apps/tzshlib/tzsh_netinit.c:761: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxE)
apps/tzshlib/tzsh_netinit.c:763: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxE)
os/arch/arm/include/armv7-m/syscall.h:258: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxV)
os/arch/arm/include/armv7-m/syscall.h:259: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:ExV)
os/arch/arm/include/armv7-m/syscall.h:260: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:ExV)
os/arch/arm/include/imxrt/imxrt105x_irq.h:496: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/include/imxrt/imxrt105x_irq.h:496: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/include/imxrt/imxrt105x_irq.h:497: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/include/imxrt/imxrt105x_irq.h:497: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/include/spinlock.h:69: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/include/spinlock.h:72: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/include/spinlock.h:73: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/include/spinlock.h:76: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/armv7-m/cache.h:343: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/arch/arm/src/armv7-m/cache.h:393: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/arch/arm/src/armv7-m/cache.h:451: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/arch/arm/src/armv7-m/cache.h:451: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/arch/arm/src/armv7-m/cache.h:456: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/arch/arm/src/common/up_internal.h:140: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/common/up_internal.h:374: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/common/up_internal.h:393: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/common/up_internal.h:406: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:176: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:176: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:177: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:177: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:181: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:181: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:182: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:182: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:187: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:187: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:188: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:188: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:191: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:191: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:192: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:192: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:194: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:194: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:195: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:195: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:199: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:199: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:200: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:200: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:203: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:203: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:204: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:204: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:209: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:209: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:210: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:210: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:211: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:211: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:214: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:214: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:215: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:215: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:217: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:217: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:218: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:218: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:222: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:222: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:223: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:223: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:628: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:628: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:629: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:629: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:633: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:633: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:634: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:634: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:644: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:644: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:648: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:648: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:653: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:653: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:656: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:656: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:661: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:661: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:714: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/chip/imxrt105x_pinmux.h:714: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/chip/imxrt_ccm.h:616: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/arch/arm/src/imxrt/chip/imxrt_gpio.h:103: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/arch/arm/src/imxrt/chip/imxrt_lpi2c.h:301: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/chip/imxrt_lpi2c.h:301: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/chip/imxrt_lpuart.h:319: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/chip/imxrt_lpuart.h:319: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/chip/imxrt_lpuart.h:320: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/chip/imxrt_lpuart.h:320: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/chip/imxrt_lpuart.h:321: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/chip/imxrt_lpuart.h:321: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/chip/imxrt_snvs.h:182: ERROR: [SPC_M_SEP] space prohibited between function name and open parenthesis '('
os/arch/arm/src/imxrt/imxrt_edma.h:412: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/arch/arm/src/imxrt/imxrt_edma.h:426: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/arch/arm/src/imxrt/imxrt_edma.h:426: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/arch/arm/src/imxrt/imxrt_enet.c:249: ERROR: [SPC_M_SEP] space prohibited before that close parenthesis ')'
os/arch/arm/src/imxrt/imxrt_enet.c:409: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxV)
os/arch/arm/src/imxrt/imxrt_enet.c:410: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:ExV)
os/arch/arm/src/imxrt/imxrt_enet.c:419: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxV)
os/arch/arm/src/imxrt/imxrt_enet.c:420: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:ExV)
os/arch/arm/src/imxrt/imxrt_gpio.h:365: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/arch/arm/src/imxrt/imxrt_lpi2c.c:130: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/arch/arm/src/imxrt/imxrt_lpi2c.c:131: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/arch/arm/src/imxrt/imxrt_lpi2c.c:131: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/arch/arm/src/imxrt/imxrt_mpuinit.c:76: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/arch/arm/src/imxrt/imxrt_mpuinit.c:80: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/arch/arm/src/imxrt/imxrt_mpuinit.h:113: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/arch/arm/src/imxrt/imxrt_serial.c:344: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/arm/src/imxrt/imxrt_serial.c:345: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/arm/src/imxrt/imxrt_serial.c:346: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/arm/src/imxrt/imxrt_start.c:317: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
os/arch/arm/src/imxrt/imxrt_start.c:317: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
os/arch/arm/src/imxrt/imxrt_start.c:317: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxB)

**Comment by seinfra** [comment]
Target : [cc850fca975972b86012fd6c6d586c52a30603d3] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [cc850fca975972b86012fd6c6d586c52a30603d3] - Code Rule Check OK.

**Comment by Ellab-Yoon** [comment]
My toolchain doesn't make any error with the 'QEMU' hardware in my linux box.
So I have no idea what to do.

- Ubuntu 14.04 /  gcc-arm-none-eabi-4_9-2015q3
and
- Kali Rolling 18.04 / gcc-arm-none-eabi-4_9-2015q3

**Comment by sunghan-chang** [comment]
@Ellab-Yoon Hello Yoon. Thank you for contribution. I will be back with feedback.

**Comment by jeongchanKim** [comment]
The build fail is about different path for up_XXX.s files.
`arch/arm/src/armv7-m -> arch/arm/src/armv7-m/gnu`

And after build,

File: os/kernel/sched/sched_yield.c [hunk 1]
```diff
@@ -130,7 +130,7 @@ int sched_yield(void)
 		switch_needed = true;
 
 		/* Remove the TCB from the ready-to-run list */
-		dq_rem((FAR dq_entry_t *)rtcb, (FAR dq_queue_t *)&g_readytorun);
+		dq_rem((FAR dq_entry_t *) rtcb, (FAR dq_queue_t *)&g_readytorun);
```
Comment (jeongchanKim) [inline os/kernel/sched/sched_yield.c]: please remove space after type casting.

File: os/kernel/sched/sched_waitpid.c [hunk 2]
```diff
@@ -342,7 +342,7 @@ pid_t waitpid(pid_t pid, int *stat_loc, int options)
 	if (rtcb->group->tg_children == NULL && retains) {
 		err = ECHILD;
 		goto errout_with_errno;
-	} else if (pid != (pid_t)-1) {
+	} else if (pid != (pid_t) - 1) {
```
Comment (jeongchanKim) [inline os/kernel/sched/sched_waitpid.c]: please remove spaces before and after '-', because this '-' doesn't mean minus.

File: os/kernel/sched/sched_waitpid.c [hunk 3]
```diff
@@ -374,7 +374,7 @@ pid_t waitpid(pid_t pid, int *stat_loc, int options)
 
 		err = ECHILD;
 		goto errout_with_errno;
-	} else if (pid != (pid_t)-1) {
+	} else if (pid != (pid_t) - 1) {
```
Comment (jeongchanKim) [inline os/kernel/sched/sched_waitpid.c]: same as above.

File: os/kernel/sched/sched_waitpid.c [hunk 4]
```diff
@@ -400,7 +400,7 @@ pid_t waitpid(pid_t pid, int *stat_loc, int options)
 		 * instead).
 		 */
 
-		if (pid == (pid_t)-1) {
+		if (pid == (pid_t) - 1) {
```
Comment (jeongchanKim) [inline os/kernel/sched/sched_waitpid.c]: same as above.

File: os/kernel/sched/sched_waitpid.c [hunk 5]
```diff
@@ -471,7 +471,7 @@ pid_t waitpid(pid_t pid, int *stat_loc, int options)
 		 * instead).
 		 */
 
-		if (rtcb->nchildren == 0 || (pid != (pid_t)-1 && (ret = kill(pid, 0)) < 0)) {
+		if (rtcb->nchildren == 0 || (pid != (pid_t) - 1 && (ret = kill(pid, 0)) < 0)) {
```
Comment (jeongchanKim) [inline os/kernel/sched/sched_waitpid.c]: same as above.

File: os/kernel/sched/sched_waitpid.c [hunk 6]
```diff
@@ -493,7 +493,7 @@ pid_t waitpid(pid_t pid, int *stat_loc, int options)
 		 * pid == (pid_t)-1, we are waiting for any child thread.
 		 */
 
-		if (info.si_signo == SIGCHLD && (pid == (pid_t)-1 || info.si_pid == pid)) {
+		if (info.si_signo == SIGCHLD && (pid == (pid_t) - 1 || info.si_pid == pid)) {
```
Comment (jeongchanKim) [inline os/kernel/sched/sched_waitpid.c]: same as above.

File: os/kernel/sched/sched_waitid.c [hunk 7]
```diff
@@ -227,7 +227,7 @@ int waitid(idtype_t idtype, id_t id, FAR siginfo_t *info, int options)
 	} else if (idtype == P_PID) {
 		/* Get the TCB corresponding to this PID and make sure it is our child. */
 
-		ctcb = sched_gettcb((pid_t)id);
+		ctcb = sched_gettcb((pid_t) id);
```
Comment (jeongchanKim) [inline os/kernel/sched/sched_waitid.c]: please remove space after typecasting.

File: os/kernel/sched/sched_resumescheduler.c [hunk 8]
```diff
@@ -0,0 +1,146 @@
+/****************************************************************************
```
Comment (jeongchanKim) [inline os/kernel/sched/sched_resumescheduler.c]: Please add samsung license.

File: os/kernel/sched/sched_mergeprioritized.c [hunk 9]
```diff
@@ -0,0 +1,189 @@
+/****************************************************************************
```
Comment (jeongchanKim) [inline os/kernel/sched/sched_mergeprioritized.c]: please add samsung license.

File: os/kernel/init/os_start.c [hunk 10]
```diff
@@ -126,6 +127,8 @@ extern bool heapx_is_init[CONFIG_MM_NHEAPS];
 
 volatile dq_queue_t g_readytorun;
 
+FAR struct tcb_s *g_running_tasks[1];
```
Comment (jeongchanKim) [inline os/kernel/init/os_start.c]: what is the usage of this variable?

File: os/arch/arm/src/imxrt/imxrt_allocateheap.c [hunk 11]
```diff
@@ -0,0 +1,395 @@
+/****************************************************************************
+ *
+ * Copyright 2016 Samsung Electronics All Rights Reserved.
```
Comment (jeongchanKim) [inline os/arch/arm/src/imxrt/imxrt_allocateheap.c]: change the license year to 2018

File: os/kernel/init/os_start.c (hunk 10 above)
Comment (Ellab-Yoon) [reply os/kernel/init/os_start.c]: It's not in use.

**Comment by seinfra** [comment]
Target : [193f19bc2a3361ce43bd869d43d7719fdc7a702c] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [193f19bc2a3361ce43bd869d43d7719fdc7a702c] - Code Rule Check OK.

**Comment by jeongchanKim** [comment]
> > The build fail is from different path for up_XXX.s files.
> > `arch/arm/src/armv7-m -> arch/arm/src/armv7-m/gnu`
> > Please check the path for armv7-m.
> 
> @jeongchanKim Can I get the entire compilation messages you mentioned?

It is as same as travis-ci result.
```
make[1]: Entering directory '/root/tizenrt/os/arch/arm/src'
make[1]: *** No rule to make target 'up_saveusercontext.S', needed by 'up_saveusercontext.o'.  Stop.
make[1]: Leaving directory '/root/tizenrt/os/arch/arm/src'
LibTargets.mk:255: recipe for target 'arch/arm/src/libarch.a' failed
make: *** [arch/arm/src/libarch.a] Error 2
```

File: apps/Application.mk [hunk 12]
```diff
@@ -0,0 +1,258 @@
+############################################################################
```
Comment (yashwanth686007) [inline apps/Application.mk]: Add samsung license for this file too.

**Comment by seinfra** [comment]
Target : [010f48c09e5b953f75a9683ca519504920d3b3bd] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [010f48c09e5b953f75a9683ca519504920d3b3bd] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [d3d445184022069d192185b6b0f02ff5a2b28785] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [d3d445184022069d192185b6b0f02ff5a2b28785] - Code Rule Check (C++) OK.

File: apps/Application.mk [hunk 13]
```diff
@@ -0,0 +1,275 @@
+###########################################################################
+#
+# Copyright 2016 Samsung Electronics All Rights Reserved.
```
Comment (jeongchanKim) [inline apps/Application.mk]: Please change the license year to 2018.

File: apps/system/tzsh/Kconfig [hunk 14]
```diff
@@ -0,0 +1,91 @@
+#
+# For a description of the syntax of this configuration file,
+# see the file kconfig-language.txt in the NuttX tools repository.
+#
+
+config SYSTEM_TZSH
```
Comment (jeongchanKim) [inline apps/system/tzsh/Kconfig]: TizenRT already supported shell(TASH). Why did you add new shell?

File: build/configs/artik053/grpc/defconfig [hunk 15]
```diff
@@ -421,7 +421,8 @@ CONFIG_NETDEV_MULTINIC=y
 # CONFIG_NET_E1000 is not set
 # CONFIG_NET_SLIP is not set
 # CONFIG_NET_VNET is not set
-# CONFIG_PIPES is not set
+CONFIG_PIPES=y
```
Comment (jeongchanKim) [inline build/configs/artik053/grpc/defconfig]: Why did you change this configuration?

File: build/configs/imxrt1050-evk/include/board.h [hunk 16]
```diff
@@ -0,0 +1,242 @@
+/****************************************************************************
```
Comment (jeongchanKim) [inline build/configs/imxrt1050-evk/include/board.h]: This file should be placed in os/arch/arm/include

File: build/configs/imxrt1050-evk/kernel/.gitignore [hunk 17]
```diff
@@ -0,0 +1,2 @@
+/nuttx_user.elf
```
Comment (jeongchanKim) [inline build/configs/imxrt1050-evk/kernel/.gitignore]: This is not needed.

File: external/wpa_supplicant/src/utils/eloop.c [hunk 18]
```diff
@@ -10,7 +10,7 @@
 #include <assert.h>
 #include <sys/select.h>
 #ifdef CONFIG_OS_NUTTX
-#include <nuttx/arch.h>
+#include <tinyara/arch.h>
```
Comment (jeongchanKim) [inline external/wpa_supplicant/src/utils/eloop.c]: Please revert this change, because the condition is for CONFIG_OS_NUTTX.

File: framework/src/iotbus/Kconfig [hunk 19]
```diff
@@ -5,43 +5,43 @@
 
 config IOTBUS
 	bool "IoTbus framework Support"
-	default y
+	default n
```
Comment (jeongchanKim) [inline framework/src/iotbus/Kconfig]: Why did you change this default config setting?

File: lib/libc/fixedmath/Make.defs [hunk 20]
```diff
@@ -1,6 +1,6 @@
 ###########################################################################
 #
-# Copyright 2016 Samsung Electronics All Rights Reserved.
+# Copyright 2018 Samsung Electronics All Rights Reserved.
```
Comment (jeongchanKim) [inline lib/libc/fixedmath/Make.defs]: Please revert license year to 2016, because it is not the last modified year, but the first written year.

File: lib/libc/fixedmath/lib_fixedmath.c [hunk 21]
```diff
@@ -1,6 +1,6 @@
 /****************************************************************************
  *
- * Copyright 2016 Samsung Electronics All Rights Reserved.
+ * Copyright 2018 Samsung Electronics All Rights Reserved.
```
Comment (jeongchanKim) [inline lib/libc/fixedmath/lib_fixedmath.c]: Please revert this.

File: lib/libc/inttypes/lib_strtoimax.c [hunk 22]
```diff
@@ -110,14 +117,14 @@ intmax_t strtoimax(FAR const char *nptr, FAR char **endptr, int base)
 		/* Correct the sign of the result and check for overflow */
 
 		if (negate) {
-			const uintmax_t limit = ((uintmax_t) - (INTMAX_MIN + 1)) + 1;
+			const uintmax_t limit = ((uintmax_t)- (INTMAX_MIN + 1)) + 1;
```
Comment (jeongchanKim) [inline lib/libc/inttypes/lib_strtoimax.c]: Please remove space after '-'.

File: lib/libc/inttypes/lib_strtoumax.c [hunk 23]
```diff
@@ -60,7 +60,7 @@
 #include <inttypes.h>
 #include <errno.h>
 
-#include "lib_internal.h"
+#include "libc.h"
```
Comment (jeongchanKim) [inline lib/libc/inttypes/lib_strtoumax.c]: Please rever this.

File: lib/libc/libc.h [hunk 24]
```diff
@@ -0,0 +1,252 @@
+/****************************************************************************
```
Comment (jeongchanKim) [inline lib/libc/libc.h]: This file is not needed. It is almost as same as lib_internal.h.

**Comment by seinfra** [comment]
Target : [8c40b4a6011d2e0b112f8f94c0b1133970383c91] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [8c40b4a6011d2e0b112f8f94c0b1133970383c91] - Code Rule Check OK.

File: build/configs/artik053/grpc/defconfig (hunk 15 above)
Comment (Ellab-Yoon) [reply build/configs/artik053/grpc/defconfig]: I enabled the PIPES feature to avoid copilation errors.

File: apps/system/tzsh/Kconfig (hunk 14 above)
Comment (Ellab-Yoon) [reply apps/system/tzsh/Kconfig]: It is more powerful than you have.

**Comment by seinfra** [comment]
Target : [0b2dd57973536f43ee8d0e1d306183dd958de797] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [0b2dd57973536f43ee8d0e1d306183dd958de797] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [5e290d5cbb0f11bbaa8bebd6610c6797fa4b27a9] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [5e290d5cbb0f11bbaa8bebd6610c6797fa4b27a9] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [8dd432961bf9ba86ec38b3371184fc2b31eb57ed] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [8dd432961bf9ba86ec38b3371184fc2b31eb57ed] - Code Rule Check (C++) OK.
